### PR TITLE
BHBC-1241: Validation Endpoint

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1241

## This PR contains the following changes

- Update validate endpoint

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/paths/dwc/validate.ts [hunk 1]
```diff
@@ -161,26 +77,78 @@ POST.apiDoc = {
...
+
+      if (!sqlStatement) {
+        throw new HTTP400('Failed to build SQL get statement');
+      }
+
+      await connection.open();
+
+      const response = await connection.query(sqlStatement.text, sqlStatement.values);
+
+      if (!response || !response.rowCount) {
```
Comment (sdevalapurkar) [inline api/src/paths/dwc/validate.ts]: we should check for existence of `response.rows` and maybe `response.rows[0]` here so we can confidently get the key from it below

File: api/src/paths/dwc/validate.ts [hunk 2]
```diff
@@ -161,26 +77,78 @@ POST.apiDoc = {
...
     try {
-      const rawMediaArray: Express.Multer.File[] = req.files as Express.Multer.File[];
+      const s3Key = req['s3Key'];
 
-      const rawMediaFile = rawMediaArray[0];
+      const s3File = await getFileFromS3(s3Key);
 
-      const mediaFiles = parseUnknownMedia(rawMediaFile);
+      if (!s3File) {
+        throw new HTTP400('Failed to get occurrence submission file');
```
Comment (sdevalapurkar) [inline api/src/paths/dwc/validate.ts]: question: here is it a 400 cause its bad request from user or should it be 500 like we failed to get the file cause internal issue?

**Review by sdevalapurkar — COMMENTED** [review]
looking good! :)

File: api/src/paths/dwc/validate.ts (hunk 1 above)
Comment (NickPhura) [reply api/src/paths/dwc/validate.ts]: I double checked what rowCount is, and my understanding had been a bit off.  rowCount is the number of affected rows, but not necessarily the number of rows returned, which makes sense.  So ya, changing this to check for rows is appropriate.

File: api/src/paths/dwc/validate.ts (hunk 2 above)
Comment (NickPhura) [reply api/src/paths/dwc/validate.ts]: Yeah I can update this.  

A bit of tech debt for the todo list is to do a general audit of all of our errors, and make sure they are informative. We can also tack on any additional error messaging as the second param in all of the HTTP errors.

**Review by sdevalapurkar — DISMISSED** [review]

**Review by sdevalapurkar — APPROVED** [review]

**Pull request merged**